src/reducers/fileTree.tsx

```diff
@@ -372,6 +376,75 @@ export const goToRelativeFile = ({
   };
 };
 
+type GoToRelativeMessageParams = {
+  _getRelativeMessage?: typeof getRelativeMessage;
+  _viewVersionFile?: typeof viewVersionFile;
+  currentMessageUid: LinterMessage['uid'] | void;
+  currentPath: string;
+  messageMap: LinterMessageMap;
+  pathList: string[];
+  position: RelativePathPosition;
+  versionId: number;
+};
+
+export const goToRelativeMessage = ({
+  /* istanbul ignore next */
+  _getRelativeMessage = getRelativeMessage,
+  /* istanbul ignore next */
+  _viewVersionFile = viewVersionFile,
+  currentMessageUid,
+  currentPath,
+  messageMap,
+  pathList,
+  position,
+  versionId,
+}: GoToRelativeMessageParams): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const nextRelativeMessageInfo = _getRelativeMessage({
+      currentMessageUid,
+      currentPath,
+      messageMap,
+      pathList,
+      position,
+    });
+
+    if (nextRelativeMessageInfo) {
+      const { line, path, uid } = nextRelativeMessageInfo;
+      const {
+        router: { location },
+      } = getState();
+
+      const queryParams = queryString.parse(location.search);
+      const newParams = { ...queryParams };
```

Comment:
It would be helpful to add a test that makes sure existing query parameters are preserved.